hr_holidays_public/models/hr_leave.py

```diff
@@ -23,37 +23,24 @@ def action_validate(self):
                 leave = leave.with_context(
                     employee_id=leave.employee_id.id, exclude_public_holidays=True
                 )
-            super(HrLeave, leave).action_validate()
+            super(HrLeave, leave).action_validate(check_state=check_state)
         return True
 
-    def _get_duration(self, check_leave_type=True, resource_calendar=None):
-        if self.holiday_status_id.exclude_public_holidays or not self.holiday_status_id:
-            instance = self.with_context(
-                employee_id=self.employee_id.id, exclude_public_holidays=True
+    def _get_durations(self, check_leave_type=True, resource_calendar=None):
+        for leave in self:
+            if (
+                leave.holiday_status_id.exclude_public_holidays
+                or not leave.holiday_status_id
+            ):
+                leave = leave.with_context(
+                    employee_id=leave.employee_id.id, exclude_public_holidays=True
+                )
+            return super(HrLeave, leave)._get_durations(
+                check_leave_type=check_leave_type, resource_calendar=resource_calendar
```

Comment:
Hi @xaviedoanhduy 

I know that this PR is merged but can you give me a bit of context of why you changed this please?

It causes errors because you are doing a return inside a for loop, hence returning the `_get_durations` result for only one record, not the whole recordset!

I would like to fix it asap (and I can do it) but if you have a bit of context about why you did this change, I would really appreciate.

Reply:
https://github.com/OCA/hr-holidays/pull/194

Reply:
Hi @victoralmau , thanks a lot for the fix, I'm going to have a look.
@xaviedoanhduy `_get_durations` method is called in `_compute_duration` compute method on `hr.leave` model. 
It must return a dictionary whose keys are leave ids, and Odoo expects every leave of the recordset to have an entry in the dictionary. 
In this addon you extend the `_get_durations` method but in some cases you return after only 1 record, hence the dictionary is not fully completed. It can cause errors in many flows, I'm surprised you didn't encounter. 
More generally, an extension of a method should never change the format / specs of the output of the original method.